### PR TITLE
Bugfix/float3 operators

### PR DESCRIPTION
**Pull request by marcorod94**
What's new?

- RayCast now detects hits with all elements despite the collision detect state
- Creation of == and != operators

### CODE REVIEW

**Review by Dartemiss — APPROVED** [review]
Cool! Useful implementation of the operators == and !=. :D

File: Engine/Component/ComponentCollider.cpp [hunk 1]
```diff
@@ -423,14 +423,11 @@ CollisionInformation ComponentCollider::RaycastClosestHit(float3& start, float3&
 	size_t num_hits = raycallback.m_hitPointWorld.size();
 	for (size_t i = 0; i < num_hits; ++i)
 	{
-		if (raycallback.m_collisionObjects[i]->hasContactResponse())
```
Comment (JorxSS) [inline Engine/Component/ComponentCollider.cpp:426]: No longer need it this comprobation?

File: Engine/Component/ComponentCollider.cpp [hunk 2]
```diff
@@ -449,13 +446,9 @@ std::vector<CollisionInformation> ComponentCollider::RaycastAllHits(float3& star
 	for (size_t i = 0; i < num_hits; ++i)
 	{
 		CollisionInformation collision;
-		if (raycallback.m_collisionObjects[i]->hasContactResponse())
```
Comment (JorxSS) [inline Engine/Component/ComponentCollider.cpp:452]: same question as before

**Review by JorxSS — APPROVED** [review]
GJ mi pana! Here you have my approval
image

**Pull request merged**